src/containers/ConversionForm/index.tsx

```diff
@@ -55,15 +56,35 @@ const ConversionForm = ({
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
+    const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
+    const toggleModal = () => {
+        setFileTypeErrorModalOpen(!fileTypeErrorModalOpen);
+    };
     const toggleProcessing = () => {
         setIsProcessing(!isProcessing);
     };
 
+    const validateFileType = () => {
+        if (fileToConvert) {
+            const fileExtension =
+                fileToConvert.name.split(".").pop()?.toLowerCase() || "";
+            const validExtensions = ["smoldyn"];
```

Comment:
I'm pretty sure smoldyn files are `.txt` not `.smoldyn`. @blairlyons?

Reply:
yeah some of the other engines have their own extensions. But we definitely want people to be able to convert smoldyn files, which are txt files, so that's what this needs to be. I don't think there are any files that are `.smoldyn`